karyon3-archaius2/src/main/java/com/netflix/karyon/archaius/Ec2ServerContextConfigSeeder.java

```diff
@@ -13,27 +13,26 @@
     public Config get(Config config) {
         return MapConfig.builder()
             // Amazon specific metadata
-            .put(ServerContext.DOMAIN,          config.getString("EC2_DOMAIN", ""))
-            .put(ServerContext.HOSTNAME,        config.getString("EC2_HOSTNAME", ""))
-            .put(ServerContext.PUBLIC_HOSTNAME, config.getString("EC2_PUBLIC_HOSTNAME", ""))
-            .put(ServerContext.PUBLIC_IPV4,     config.getString("EC2_PUBLIC_IPV4", ""))
-            .put(ServerContext.LOCAL_HOSTNAME,  config.getString("EC2_LOCAL_HOSTNAME", ""))
-            .put(ServerContext.LOCAL_IPV4,      config.getString("EC2_LOCAL_IPV4", ""))
-            .put(ServerContext.DATACENTER,      config.getString("EC2_REGION", ""))
-            .put(ServerContext.RACK,            config.getString("EC2_AVAILABILITY_ZONE", ""))
-            .put(ServerContext.SERVER_ID,       config.getString("EC2_INSTANCE_ID", ""))
-            .put(ServerContext.AMI,             config.getString("EC2_AMI_ID", ""))
+            .put(ServerContext.DOMAIN,          "${EC2_DOMAIN}")
+            .put(ServerContext.HOSTNAME,        "${EC2_HOSTNAME}")
+            .put(ServerContext.PUBLIC_HOSTNAME, "${EC2_PUBLIC_HOSTNAME}")
+            .put(ServerContext.PUBLIC_IPV4,     "${EC2_PUBLIC_IPV4}")
+            .put(ServerContext.LOCAL_HOSTNAME,  "${EC2_LOCAL_HOSTNAME}")
+            .put(ServerContext.LOCAL_IPV4,      "${EC2_LOCAL_IPV4}")
+            .put(ServerContext.DATACENTER,      "cloud")
+            .put(ServerContext.ZONE,            "${EC2_AVAILABILITY_ZONE}")
+            .put(ServerContext.SERVER_ID,       "${EC2_INSTANCE_ID}")
+            .put(ServerContext.AMI,             "${EC2_AMI_ID}")
+            .put(ServerContext.REGION,          "${EC2_REGION}")
             
             // Netflix specific metadata
-            .put(ServerContext.ENVIRONMENT,     config.getString("NETFLIX_ENVIRONMENT", ""))
-            .put(ServerContext.CLUSTER,         config.getString("NETFLIX_CLUSTER", ""))
-            .put(ServerContext.ASG,             config.getString("NETFLIX_AUTO_SCALE_GROUP", ""))
-            .put(ServerContext.STACK,           config.getString("NETFLIX_STACK", ""))
-            .put(ServerContext.APP_ID,          config.getString("NETFLIX_APP", ""))
+            .put(ServerContext.ENVIRONMENT,     "${NETFLIX_ENVIRONMENT}")
```

Comment:
I don't see the harm.

The EC2_* environment variables are also a Netflix convention based on how we setup the baseami and user-data. EC2 prefixed variables come from the metadata service on instance but get added as environment variables by scripts in our baseami. NETFLIX prefixed ones are part of the user-data that Asgard generates (though Asgard allows a configurable prefix). So it could potentially be useful to anyone deploying using Asgard.

It seems like this would help make it easier for others using the Netflix open source tools to run using a setup like we actually use internally. The user could always provide a custom ConfigSeeder if they want to change the behavior. Also having an idea of how we setup these components might be useful to others trying to figure out how to piece it all together.